app/models/package_manager/cpan.rb

```diff
@@ -46,19 +46,19 @@ def self.mapping(raw_project)
     end
 
     def self.versions(raw_project, _name)
-      versions = get("https://fastapi.metacpan.org/v1/release/_search?q=distribution:#{raw_project['distribution']}&size=5000&fields=version,date")["hits"]["hits"]
+      versions = get("https://fastapi.metacpan.org/v1/release/_search?q=distribution:#{raw_project['distribution']}&size=5000&_source=version,date")["hits"]["hits"]
       versions.map do |version|
         {
-          number: version["fields"]["version"],
-          published_at: version["fields"]["date"],
+          number: version["_source"]["version"],
+          published_at: version["_source"]["date"],
         }
       end
     end
 
     def self.dependencies(_name, version, mapped_project)
-      versions = mapped_project[:versions]
-      version_data = versions.find { |v| v["fields"]["version"] == version }
-      version_data["fields"]["dependency"].select { |dep| dep["relationship"] == "requires" }.map do |dep|
+      versions = get("https://fastapi.metacpan.org/v1/release/_search?q=distribution:#{mapped_project[:name]}&size=5000&_source=version,dependency")["hits"]["hits"]
```

Comment:
we're also passing in `mapped_project` here, which only has `number` and `published_at`, so we have to make an extra request to get the dependencies.